# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorBridgeParent.cpp
# Commit: 156e568ecf00
# Full Hash: 156e568ecf00d47c0dbe0530d5696b612f391831
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2023-05-12 04:06:42
# Description:
#   Bug 1827340 - Update WebRender settings only from root WebRenderBridgeParent r=lsalzman
#   
#   WebRenderAPI::UpdateDebugFlags() call from non-root WebRenderBridgeParent is not necessary.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D177454
# ==============================================================================

diff -r c132cd7a5920 -r 156e568ecf00 gfx/layers/ipc/CompositorBridgeParent.cpp
--- a/gfx/layers/ipc/CompositorBridgeParent.cpp	Fri May 12 00:38:58 2023 +0000
+++ b/gfx/layers/ipc/CompositorBridgeParent.cpp	Fri May 12 00:53:27 2023 +0000
@@ -1232,6 +1232,9 @@
 
   MonitorAutoLock lock(*sIndirectLayerTreesLock);
   ForEachWebRenderBridgeParent([&](WebRenderBridgeParent* wrBridge) -> void {
+    if (!wrBridge->IsRootWebRenderBridgeParent()) {
+      return;
+    }
     wrBridge->UpdateQualitySettings();
   });
 }
@@ -1252,6 +1255,9 @@
 
   MonitorAutoLock lock(*sIndirectLayerTreesLock);
   ForEachWebRenderBridgeParent([&](WebRenderBridgeParent* wrBridge) -> void {
+    if (!wrBridge->IsRootWebRenderBridgeParent()) {
+      return;
+    }
     wrBridge->UpdateDebugFlags();
   });
 }
@@ -1270,6 +1276,9 @@
 
   MonitorAutoLock lock(*sIndirectLayerTreesLock);
   ForEachWebRenderBridgeParent([&](WebRenderBridgeParent* wrBridge) -> void {
+    if (!wrBridge->IsRootWebRenderBridgeParent()) {
+      return;
+    }
     wrBridge->UpdateBoolParameters();
   });
 }
@@ -1288,6 +1297,9 @@
 
   MonitorAutoLock lock(*sIndirectLayerTreesLock);
   ForEachWebRenderBridgeParent([&](WebRenderBridgeParent* wrBridge) -> void {
+    if (!wrBridge->IsRootWebRenderBridgeParent()) {
+      return;
+    }
     wrBridge->UpdateParameters();
   });
 }
@@ -1299,6 +1311,9 @@
   }
   MonitorAutoLock lock(*sIndirectLayerTreesLock);
   ForEachWebRenderBridgeParent([&](WebRenderBridgeParent* wrBridge) -> void {
+    if (!wrBridge->IsRootWebRenderBridgeParent()) {
+      return;
+    }
     wrBridge->UpdateProfilerUI();
   });
 }